custom_components/ramses_cc/binary_sensor.py

```diff
@@ -39,124 +80,214 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType = None,
 ) -> None:
-    """Create binary sensors for CH/DHW (heat) & HVAC.
-
-    discovery_info keys:
-      gateway: is the ramses_rf protocol stack (gateway/protocol/transport/serial)
-      devices: heat (e.g. CTL, OTB, BDR, TRV) or hvac (e.g. FAN, CO2, SWI)
-      domains: TCS, DHW and Zones
-    """
+    """Create binary sensors for CH/DHW (heat) & HVAC."""
 
     if discovery_info is None:
         return
 
     broker: RamsesBroker = hass.data[DOMAIN][BROKER]
 
-    def entity_factory(
-        broker: RamsesBroker, device, attr, *, entity_class=None, **kwargs
-    ):
-        return (entity_class or RamsesBinarySensor)(broker, device, attr, **kwargs)
-
-    new_sensors = [
-        entity_factory(broker, dev, k, **v)
-        for k, v in BINARY_SENSOR_ATTRS["gateway"].items()
-        if (dev := discovery_info.get("gateway"))
-    ]  # 18:xxxxxx - status
-    new_sensors += [
-        entity_factory(broker, dev, k, **v)
-        for key in ("devices", "domains")
-        for dev in discovery_info.get(key, [])
-        for k, v in BINARY_SENSOR_ATTRS[key].items()
-        if dev and hasattr(dev, k)
+    sensor_types: tuple[RamsesBinarySensorEntityDescription, ...] = (
+        RamsesBinarySensorEntityDescription(
+            key="gateway",
+            name="Gateway",
+            rf_class=Gateway,
+            entity_class=RamsesGatewayBinarySensor,
+            device_class=BinarySensorDeviceClass.PROBLEM,
+        ),
+        RamsesBinarySensorEntityDescription(
+            key="system",
+            name="System",
+            rf_class=System,
+            entity_class=RamsesSystemBinarySensor,
+            device_class=BinarySensorDeviceClass.PROBLEM,
+            extra_attributes={
+                ATTR_WORKING_SCHEMA: SZ_SCHEMA,
```

Comment:
Just noticed that this changed to `working_schema` recently so have updated accordingly.

Reply:
Yes - schemas are a real challenge.

Lots have people have difficulty with schema, despite my best efforts - a sign it isn't implemented well.

There have been lots of recent changes with warnings in logs, improvements to the wiki, and others like this: renaming the attr to make it clear it is the active schema (and not the cached, config, minimal schema).

Reply:
> Just noticed that this changed to working_schema recently so have updated accordingly.

There have been changes to **master** that **rafactor** should have kept up with - I did look for one, and found it (removing `oem_code` for all but OTBs), so was happy. Can I leave that with you - I won't make any more big changes without porting across myself.